test/EFCore.PG.FunctionalTests/Query/NorthwindFunctionsQueryNpgsqlTest.cs

```diff
@@ -401,6 +402,165 @@ public virtual async Task GroupBy_JsonbAgg(bool async)
+                        g.Select(c => ValueTuple.Create(c.CompanyName, c.ContactName)))
+                });
+
+        var results = async
+            ? await query.ToListAsync()
+            : query.ToList();
+
+        var london = results.Single(r => r.City == "London");
+
+        Assert.Equal(
+            new Dictionary<string, string>
+            {
+                ["Around the Horn"] = "Thomas Hardy",
+                ["B's Beverages"] = "Victoria Ashworth",
+                ["Consolidated Holdings"] = "Elizabeth Brown",
+                ["Eastern Connection"] = "Ann Devon",
+                ["North/South"] = "Simon Crowther",
+                ["Seven Seas Imports"] = "Hari Kumar"
+            },
+            london.Companies);
+
+        AssertSql(
+            @"SELECT c.""City"", json_object_agg(c.""CompanyName"", c.""ContactName"") AS ""Companies""
+FROM ""Customers"" AS c
+GROUP BY c.""City""");
+    }
+
+    [ConditionalTheory]
+    [MemberData(nameof(IsAsyncData))]
+    public virtual async Task GroupBy_JsonbObjectAgg(bool async)
+    {
+        using var ctx = CreateContext();
+
+        // Note that unlike with json, jsonb doesn't guarantee ordering; so we parse the JSON string client-side.
+        var query = ctx.Set<Customer>()
+            .GroupBy(c => c.City)
+            .Select(
+                g => new
+                {
+                    City = g.Key,
+                    Companies = EF.Functions.JsonbObjectAgg(g.Select(c => ValueTuple.Create(c.CompanyName, c.ContactName)))
+                });
+
+        var results = async
+            ? await query.ToListAsync()
+            : query.ToList();
+
+        var london = results.Single(r => r.City == "London");
+        var companiesDictionary = JsonSerializer.Deserialize<Dictionary<string, string>>(london.Companies);
+
+        Assert.Equal(
+            new Dictionary<string, string>
+            {
+                ["Around the Horn"] = "Thomas Hardy",
+                ["B's Beverages"] = "Victoria Ashworth",
+                ["Consolidated Holdings"] = "Elizabeth Brown",
+                ["Eastern Connection"] = "Ann Devon",
+                ["North/South"] = "Simon Crowther",
+                ["Seven Seas Imports"] = "Hari Kumar"
+            },
+            companiesDictionary);
+
+        AssertSql(
+            @"SELECT c.""City"", jsonb_object_agg(c.""CompanyName"", c.""ContactName"") AS ""Companies""
+FROM ""Customers"" AS c
+GROUP BY c.""City""");
+    }
+
+    [ConditionalTheory]
+    [MemberData(nameof(IsAsyncData))]
+    public virtual async Task GroupBy_JsonbObjectAgg_as_Dictionary(bool async)
+    {
+        using var ctx = CreateContext();
+
+        var query = ctx.Set<Customer>()
+            .GroupBy(c => c.City)
+            .Select(
+                g => new
+                {
+                    City = g.Key,
+                    Companies = EF.Functions.JsonbObjectAgg<string, string, Dictionary<string, string>>(
```

Comment:
And probably my last aggregate work for now :relieved: 

We now support the PG [jsonb_object_agg](https://www.postgresql.org/docs/current/functions-aggregate.html#FUNCTIONS-AGGREGATE-TABLE) function, using it to project out a `Dictionary<string,string>` out of two arbitrary columns in a table. Can also get a string JSON document (see test above).

(@ajcvickers this goes in the direction of projecting raw JSON data so we can send it directly over HTTP without entity dematerialization).

/cc @smitpatel @maumar @ajcvickers